src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -292,5 +309,127 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
             result = results;
             return true;
         }
+
+        internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            if (!TryReadFile(statusFilePath, reusableReader, out string fileContents))
+            {
+                // Between the time that we get an ID and the time that we try to read the associated stat
+                // file(s), the process could be gone.
+                result = default(ParsedStatus);
+                return false;
+            }
+
+            ParsedStatus results = default(ParsedStatus);
+            ReadOnlySpan<char> statusFileContents = fileContents.AsSpan();
+            int unitSliceLength = -1;
+#if DEBUG
+            int nonUnitSliceLength = -1;
+#endif
+            while (!statusFileContents.IsEmpty)
+            {
+                int startIndex = statusFileContents.IndexOf(':');
+                if (startIndex == -1)
+                {
+                    // Reached end of file
+                    break;
+                }
+
+                ReadOnlySpan<char> title = statusFileContents.Slice(0, startIndex);
+                statusFileContents = statusFileContents.Slice(startIndex + 1);
+                int endIndex = statusFileContents.IndexOf('\n');
+                if (endIndex == -1)
+                {
+                    endIndex = statusFileContents.Length - 1;
+                    unitSliceLength = statusFileContents.Length - 3;
+#if DEBUG
+                    nonUnitSliceLength = statusFileContents.Length;
+#endif
+                }
+                else
+                {
+                    unitSliceLength = endIndex - 3;
+#if DEBUG
+                    nonUnitSliceLength = endIndex;
+#endif
+                }
+
+                ReadOnlySpan<char> value = default;
```

Comment:
Nit: this local is unnecessary... instead of e.g.
```C#
value = statusFileContents.Slice(0, nonUnitSliceLength);
valueParsed = int.TryParse(value, out results.Pid);
```
it can be:
```C#
valueParsed = int.TryParse(statusFileContents.Slice(0, nonUnitSliceLength), out results.Pid);
```